ambry-store/src/test/java/com.github.ambry.store/IndexTest.java

```diff
@@ -1237,6 +1243,39 @@ public void getIndexSegmentFilesForLogSegmentTest() {
     }
   }
 
+  /**
+   * Tests {@link PersistentIndex#close()} can correctly cancel the scheduled persistor task and makes sure no persistor
+   * is running background after index closed.
+   * @throws StoreException
+   * @throws InterruptedException
+   */
+  @Test
+  public void closeIndexToCancelPersistorTest() throws StoreException, InterruptedException {
+    state.index.close();
+    // re-initialize index by using mock scheduler (the intention is to speed up testing by using shorter period)
+    ScheduledThreadPoolExecutor scheduler = (ScheduledThreadPoolExecutor) Utils.newScheduler(1, false);
+    ScheduledThreadPoolExecutor mockScheduler = Mockito.spy(scheduler);
+    AtomicReference<ScheduledFuture> persistorTask = new AtomicReference<>();
+    MockIndexPersistor mockPersistor = new MockIndexPersistor();
+    mockPersistor.invokeCountDown = new CountDownLatch(1);
+    doAnswer(invocation -> {
+      Object[] args = invocation.getArguments();
+      persistorTask.set(mockScheduler.scheduleAtFixedRate(mockPersistor, 0, 50, TimeUnit.MILLISECONDS));
+      return persistorTask.get();
+    }).when(mockScheduler)
+        .scheduleAtFixedRate(any(Runnable.class), anyLong(), anyLong(), any(TimeUnit.SECONDS.getClass()));
+    state.initIndex(mockScheduler);
+    // verify that the persistor task is successfully scheduled
+    assertTrue("The persistor task wasn't invoked within the expected time",
+        mockPersistor.invokeCountDown.await(10, TimeUnit.MILLISECONDS));
+    state.index.close();
+    mockPersistor.invokeCountDown = new CountDownLatch(1);
+    // verify that the persisitor task is canceled after index closed and is never invoked again.
+    assertTrue("The persistor task should be canceled after index closed", persistorTask.get().isCancelled());
+    assertFalse("The persistor task should not be invoked again",
```

Comment:
looks like previous assert `assertTrue("The persistor task should be canceled after index closed", persistorTask.get().isCancelled())` will suffice. However, based on my experiment, `isCancelled` only checks the internal flag of the task, the scheduled task in the queue still would proceed. This assert ensures the task queue is also empty because in no more task is invoked after a period of time.

Reply:
Discussed offline about making the timeout shorter and using the same constant when scheduling and here